common/lib/xmodule/xmodule/x_module.py

```diff
@@ -1743,6 +1744,37 @@ def resource_url(self, resource):
     def publish(self, block, event_type, event):
         pass
 
+    def service(self, block, service_name):
+        """
+        Runtime-specific override for the XBlock service manager.  If a service is not currently
+        instantiated and is declared as a critical requirement, an attempt is made to load the
+        module.
+
+        Arguments:
+            block (an XBlock): this block's class will be examined for service
+                decorators.
+            service_name (string): the name of the service requested.
+
+        Returns:
+            An object implementing the requested service, or None.
+        """
+        declaration = block.service_declaration(service_name)
+        if declaration is None:
+            raise NoSuchServiceError("Service {!r} was not requested.".format(service_name))
+
+        service = self._services.get(service_name)
+        if service is None and declaration == "need":
+            from xmodule.modulestore.django import ModuleI18nService
```

Comment:
@mattdrayer what do u thing If we take the advantage of parent method instead of repeating the same code ?